pkg/virtual/syncer/transformations/defaultsummarizing.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transformations
+
+import (
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var _ SummarizingRules = (*DefaultSummarizingRules)(nil)
+var _ SummarizingRulesProvider = (*DefaultSummarizingRules)(nil)
+
+// DefaultSummarizingRules provides a default minimal implementation of [SummarizingRules].
+// It only adds a status field, which for now is always promoted (see comments below in the code)
+type DefaultSummarizingRules struct{}
+
+type field struct {
+	Path              []string
```

Comment:
what is a path. Why is it public?

Reply:
field renamed, and JSON tags added so that it is clear that the fields are public to enable JSON marshalling.